pkg/kubelet/cm/device_plugin_handler.go

```diff
@@ -202,16 +210,14 @@ func (h *DevicePluginHandlerImpl) Allocate(pod *v1.Pod, container *v1.Container,
 		}
 		ret = append(ret, resp)
 	}
+	// Checkpoints device to container allocation information.
+	err := h.writeCheckpoint()
+	if err != nil {
+		return nil, err
```

Comment:
Is the intention to fail the allocation if we fail to write the checkpoint?

Reply:
yes.